File: example-integration/ios/example integration/MockDB.swift
//
//  MockDB.swift
//  example integration
//
//  Created by admin on 11.08.23.
//

import Foundation

@objc
class MockDB: NSObject {
    private static var dbInstance: MockDB? = nil
    
    @objc
    public static func instance() -> MockDB {
        if(dbInstance == nil) {
            dbInstance = MockDB()
        }
        return dbInstance!
    }
    
    private var counter: Int = 0
    
    @objc
    public func setCounter(counter: Int) {
        self.counter = counter
    }
    @objc
    public func getCounter() -> Int {
        return self.counter
    }
    
    public func increment() {
        self.counter += 1
    }
    
    public func decrement() {
        self.counter -= 1
    }
    
    public func setRandomCounter() -> Int {
        self.counter = Int.random(in: 1..<101)
        return self.counter
    }
 }

File: example-integration/android/app/src/main/java/com/flodoerr/exampleintegration/MockDB.java
package com.flodoerr.exampleintegration;

import java.util.Random;

public class MockDB {

    private static MockDB dbInstance;

    public static MockDB instance() {
        if(MockDB.dbInstance == null) {
            MockDB.dbInstance = new MockDB();
        }
        return MockDB.dbInstance;
    }

    private MockDB() { }

    private int counter = 0;

    public void setCounter(int counter) {
        this.counter = counter;
    }

    public int getCounter() {
        return counter;
    }

    public void increment() {
        counter++;
    }

    public void decrement() {
        counter--;
    }

    public int setRandomCounter() {
        counter = new Random().nextInt(100) + 1;
        return counter;
    }
}
